app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx

```diff
@@ -1,7 +1,31 @@
-import { useContext, useEffect } from 'react';
+import { useContext, useEffect, useRef, useState } from 'react';
 import AcuantContext from '../context/acuant';
 import useAsset from '../hooks/use-asset';
 import useI18n from '../hooks/use-i18n';
+import useInstanceId from '../hooks/use-instance-id';
+import useImmutableCallback from '../hooks/use-immutable-callback';
+
+/**
+ * Defines a property on the given object, calling the change callback when that property is set to
+ * a new value.
+ *
+ * @param {any} object Object on which to define property.
+ * @param {string} property Property name to observe.
+ * @param {(nextValue: any) => void} onChangeCallback Callback to trigger on change.
+ */
+export function defineObservableProperty(object, property, onChangeCallback) {
+  let currentValue;
+
+  Object.defineProperty(object, property, {
```

Comment:
random thought... likely doesn't matter... would it ever be worth checking to see if there's already a getter defined and either 1) bailing or 2) chaining the new callback?

Reply:
> random thought... likely doesn't matter... would it ever be worth checking to see if there's already a getter defined and either 1) bailing or 2) chaining the new callback?

While I'd not expect custom property getters and setters to be incredibly common, I do think a more robust implementation should check for these things, yes. For this current usage, I'm not too worried about it since we're already basing this on very open box assumptions of Acuant's implementation in how they assign to the (non-standard) element property. Ideally I'd rather we not need this at all and Acuant inform us explicitly through public APIs when "tap to capture" is triggered. But, alas...

Reply:
> > random thought... likely doesn't matter... would it ever be worth checking to see if there's already a getter defined and either 1) bailing or 2) chaining the new callback?
> 
> While I'd not expect custom property getters and setters to be incredibly common, I do think a more robust implementation should check for these things, yes. For this current usage, I'm not too worried about it since we're already basing this on very open box assumptions of Acuant's implementation in how they assign to the (non-standard) element property. Ideally I'd rather we not need this at all and Acuant inform us explicitly through public APIs when "tap to capture" is triggered. But, alas...

In practice, this could look something like we've done in a few other cases with storing an `original*` copy of a function, then overriding it and calling it with our new custom logic ([example](https://github.com/18F/identity-idp/pull/5143/files/1f8f271891e4a5b83c4928364e316cf301132ad2#diff-4c6533807dd9a52d37bbd4a31cf7d4582e9879f7d160f2e0e4beddb06231b7f6R283)).

```js
const { get: originalGet, set: originalSet } = Object.getOwnPropertyDescriptor(object, property);
```